cupy/core/_routines_statistics.pyx

```diff
@@ -8,20 +8,46 @@ from cupy.core.core cimport ndarray
 
 import cupy
 if cupy.cuda.cub_enabled:
+    import warnings
+    from cupy import util
     from cupy.cuda import cub
 
 
 cdef ndarray _ndarray_max(ndarray self, axis, out, dtype, keepdims):
     if cupy.cuda.cub_enabled:
-        if cub.can_use_reduce_max(self.dtype, self.ndim, dtype, axis):
-            return cub.reduce_max(self, out=out, keepdims=keepdims)
+        if cub.can_use_device_reduce(cub.CUPY_CUB_MAX, self.dtype, self.ndim,
+                                     axis, dtype):
+            return cub.device_reduce(self, cub.CUPY_CUB_MAX, out=out,
+                                     keepdims=keepdims)
+        elif cub.can_use_device_segmented_reduce(
+                cub.CUPY_CUB_MAX, self.dtype, self.ndim, axis, dtype):
+            if self.dtype in (numpy.complex64, numpy.complex128):
+                warnings.warn("CUB reduction for complex numbers may not be "
```

Comment:
It seems that when dim of axis > 1, and dtype is complex then is better to use the old reduce implementations,
how about not using cub in such cases?

Reply:
Note complex64 seems OK, sometimes it’s even better.